### PR TITLE
Rename fs::read_string to read_to_string and stabilize

### LINKED ISSUE #46588: Tracking issue for std::fs::read, read_string, and write
Implemented in https://github.com/rust-lang/rust/pull/45837

New APIs in `std::fs`:

```rust
pub fn read<P: AsRef<Path>>(path: P) -> io::Result<Vec<u8>> { … }
pub fn read_string<P: AsRef<Path>>(path: P) -> io::Result<String> { … }
pub fn write<P: AsRef<Path>, C: AsRef<[u8]>>(path: P, contents: C) -> io::Result<()> { ... }
```

(`read_string` is based on `read_to_string` and so returns an error on non-UTF-8 content.)

Before:

```rust
use std::fs::File;
use std::io::Read;

let mut bytes = Vec::new();
File::open(filename)?.read_to_end(&mut bytes)?;
do_something_with(bytes)
```

After:

```rust
use std::fs;

do_something_with(fs::read(filename)?)
```

### PR DESCRIPTION
**Pull request by mbrubeck**
As approved in https://github.com/rust-lang/rust/issues/46588#issuecomment-377530365

Closes #46588.

### CODE REVIEW

File: src/librustc/util/common.rs [hunk 1]
```diff
@@ -244,7 +244,8 @@ fn get_resident() -> Option<usize> {
     use std::fs;
 
     let field = 1;
-    let contents = fs::read_string("/proc/self/statm").ok()?;
+    let contents = fs::read("/proc/self/statm").ok()?;
+    let contents = String::from_utf8(contents).ok()?;
```
Comment (SimonSapin) [inline src/librustc/util/common.rs]: Why this change? Other than this r=me

Comment (mbrubeck) [reply src/librustc/util/common.rs]: Using the new `read_to_string` function here would break bootstrapping from the previous release, where that name didn't exist.  This seemed simpler than doing some sort of conditional import based on cfg(stage0).

Comment (SimonSapin) [reply src/librustc/util/common.rs]: Alright, sounds good.

**Comment by mbrubeck** [comment]
Hit the three hour limit.
